### PR TITLE
📖 Adds docs for AMP-to-AMP linking in amp-story-bookend

### LINKED ISSUE #15959: Document amp-to-amp linking in amp-story-bookend
Publishers can enable AMP-to-AMP linking by specifying `amphtml: true` on their links in the bookend.  We should document this feature in amp-story.md.

### PR DESCRIPTION
**Pull request by Enriqe**
Closes #15959

Part of #15261

### CODE REVIEW

**Comment by Enriqe** [comment]
PTAL @choumx

File: extensions/amp-story/amp-story.md [hunk 1]
```diff
@@ -446,6 +446,36 @@ src="https://github.com/ampproject/amphtml/raw/master/extensions/amp-story/img/a
   </noscript>
 </amp-img>
 
+**AMP-to-AMP linking**
+
+You can use AMP-to-AMP linking in the components that support a url by including `'amphtml': true` inside the component linking to another AMP document. For example:
```
Comment (choumx) [inline extensions/amp-story/amp-story.md]: This doesn't explain what AMP-to-AMP linking. How about:

"For documents displayed in an AMP viewer, links typically navigate `_top` or open in a new window. Links to AMP pages, however, may continue to be displayed in the viewer. To enable this behavior, add `"amphtml": true` to the link object. For example:"

/cc @bpaduch

File: extensions/amp-story/amp-story.md [hunk 2]
```diff
@@ -446,6 +446,36 @@ src="https://github.com/ampproject/amphtml/raw/master/extensions/amp-story/img/a
   </noscript>
 </amp-img>
 
+**AMP-to-AMP linking**
+
+For documents displayed in an AMP viewer, links typically navigate _top or open in a new window. Links to AMP pages, however, may continue to be displayed in the viewer. To enable this behavior, add "amphtml": true to a component that supports links. For example:
```
Comment (bpaduch) [inline extensions/amp-story/amp-story.md]: should this be "to top"?  If it's _top, use code format like `_top`

Comment (bpaduch) [inline extensions/amp-story/amp-story.md]: put "amphtml":true in code format

**Comment by Enriqe** [comment]
@bpaduch PTAL

**Pull request merged**